static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,277 @@
+          frequency,
+        },
+      });
+    },
+    [
+      actions,
+      api,
+      provider,
+      integration,
+      channel,
+      organization.features,
+      organization.slug,
+    ]
+  );
+
+  return {
+    createNotificationAction,
+    notificationProps: {
+      actions,
+      provider,
+      integration,
+      channel,
+      setActions,
+      setProvider,
+      setIntegration,
+      setChannel,
+    },
+  };
+}
+
+export default function IssueAlertNotificationOptions(
+  notificationProps: IssueAlertNotificationProps
+) {
+  const organization = useOrganization();
+  const {actions, provider, setActions, setIntegration, setProvider} = notificationProps;
+
+  const messagingIntegrationsQuery = useApiQuery<OrganizationIntegration[]>(
+    [`/organizations/${organization.slug}/integrations/?integrationType=messaging`],
+    {staleTime: Infinity}
+  );
+
+  const providersToIntegrations = useMemo(() => {
+    const map: {[key: string]: OrganizationIntegration[]} = {};
+    if (messagingIntegrationsQuery.data) {
+      for (const i of messagingIntegrationsQuery.data) {
+        const providerSlug = i.provider.slug;
+        map[providerSlug] = map[providerSlug] ?? [];
+        map[providerSlug].push(i);
+      }
+    }
+    return map;
+  }, [messagingIntegrationsQuery.data]);
+
+  useEffect(() => {
+    const providerKeys = Object.keys(providersToIntegrations);
+    if (providerKeys.length > 0) {
+      const firstProvider = providerKeys[0];
+      setProvider(firstProvider);
+
+      const firstIntegration = providersToIntegrations[firstProvider][0];
+      setIntegration(firstIntegration);
+    }
+  }, [providersToIntegrations, setProvider, setIntegration]);
+
+  const shouldRenderSetupButton = useMemo(() => {
+    return messagingIntegrationsQuery.data?.every(i => i.status !== 'active');
+  }, [messagingIntegrationsQuery.data]);
+
+  const shouldRenderNotificationConfigs = useMemo(() => {
+    return actions.some(v => v !== MultipleCheckboxOptions.EMAIL);
+  }, [actions]);
+
+  if (messagingIntegrationsQuery.isLoading || messagingIntegrationsQuery.isError) {
+    return null;
+  }
+
+  return (
+    <div>
+      <MultipleCheckbox
+        name="notification"
+        value={actions}
+        onChange={values => setActions(values)}
+      >
+        <Wrapper>
+          <MultipleCheckbox.Item value={MultipleCheckboxOptions.EMAIL} disabled>
+            Notify via email
+          </MultipleCheckbox.Item>
+          {!shouldRenderSetupButton && provider && (
+            <div>
+              <MultipleCheckbox.Item value={MultipleCheckboxOptions.INTEGRATION}>
+                Notify via integration (Slack, Discord, MS Teams, etc.)
+              </MultipleCheckbox.Item>
+              {shouldRenderNotificationConfigs && (
+                <MessagingIntegrationAlertRule
+                  notificationProps={notificationProps}
+                  providersToIntegrations={providersToIntegrations}
+                />
```

Comment:
Should this be inside of the `MultipleCheckbox` container?

Reply:
This is probably fine, I was expecting that multiple checkbox was actually rendering a list, but it looks like it's actually just divs.

If it WAS rendering a list and this looked like

```
<ul>
  <li>
    <input type="checkbox" /> Label 1
  </li>
  <li>
    <input type="checkbox" /> Notify via Integration...
  </li>
  <div>
    Send ....
  </div>
</ul>
```

This would actually be incorrect html. See

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#technical_summary

> Permitted content: `Zero or more <li>, <script> and <template> elements.`

You're not allowed to have `div` elements inside a `ul`.

That said browsers render it fine. It's just not "proper" semantic HTML.

This can have downstream effects on things like screen readers for blind users. But that 's already a problem here since we're using unsemantic HTML to render this list of check boxes (just divs) instead of using a real list.